.github/workflows/build_and_test_dbt.yaml

```diff
@@ -49,8 +64,13 @@ jobs:
       - name: Build models
         run: |
           if [[ $MODIFIED_RESOURCES_ONLY == 'true' ]]; then
-            echo "Running build on modified/new resources only"
-            dbt run -t "$TARGET" -s state:modified state:new --defer --state "$STATE_DIR"
+            if [[ $MANUALLY_DISPATCHED == 'true' ]]; then
+              echo "Running build on manually selected resources"
+              dbt run -t "$TARGET" -s ${{ inputs.models }} --defer --state "$STATE_DIR"
+            else
+              echo "Running build on modified/new resources only"
+              dbt run -t "$TARGET" -s state:modified state:new --defer --state "$STATE_DIR"
+            fi
           else
             echo "Running build on all resources"
             dbt run -t "$TARGET"
```

Comment:
**[Question, non-blocking]** I don't expect this fallback conditional branch to ever be reached, but in the rare case that it is reached, do we also need to handle manual dispatch?

```suggestion
            if [[ $MANUALLY_DISPATCHED == 'true' ]]; then
              dbt run -t "$TARGET" -s ${{ inputs.models }}
            else
              dbt run -t "$TARGET"
            fi
```

Reply:
My thinking here was:

- The only way to reach this conditional is the `master-cache` missing from S3
- In which case, we would probably want to rebuild the full DAG anyways
- Rebuilding the cache with a subset of models would lead to errors, as the workflow expects `master-cache` to encompass the full DAG

Reply:
Makes sense to me!